tests/crypto_test.py

```diff
@@ -274,8 +307,12 @@ def test_signed_only_false(self):
 
     @staticmethod
     def _context_mock():
-        error = gpgme.GpgmeError()
-        error.code = gpgme.ERR_AMBIGUOUS_NAME
+        # We need an uninittialized class that is a descendent of GPGMEError
+        # for the test, but we need to make the getcode method a mock, so we
+        # use type to make a dynamic sublcass, since raise wont' accept
+        # anything that isn't an Exception
+        error = type('CustomError', (gpg.errors.GPGMEError, ), {})
+        error.getcode = mock.Mock(return_value=gpg.errors.AMBIGUOUS_NAME)
```

Comment:
it's less typing than a local class? that's the only reason. I can change it

The problem (and the reason that those two tests failed), is that we were mocking over the original class, and that mock wasn't being unset. I'll replace it with a local class which should be clearer.